remote/remote_test.go

```diff
@@ -276,6 +278,133 @@ func TestFeeds(t *testing.T) {
 
 }
 
+func TestAccess(t *testing.T) {
+	tr, cleanup := newTestRunner(t)
+	defer cleanup()
+
+	// add datasets to both nodes
+	aRef := writeWorldBankPopulation(tr.Ctx, t, tr.NodeA.Repo)
+	bRef := writeVideoViewStats(tr.Ctx, t, tr.NodeB.Repo)
+	// establish nodeB as the client
+	cli := tr.NodeBClient(t)
+	// empty policy should allow no actions
+	rem := tr.NodeARemote(t, OptPolicy(&access.Policy{}))
+	server := tr.RemoteTestServer(rem)
+	defer server.Close()
+
+	if err := cli.PushDataset(tr.Ctx, bRef, server.URL); err.Error() != access.ErrAccessDenied.Error() {
+		t.Errorf("expected %q when trying to push dataset to a remote that does not allow pushes, got %q instead", access.ErrAccessDenied, err)
+	}
+
+	const allowPushJSON = `
+	[
+		{
+			"title": "allow subject to push its own datasets",
+			"effect": "allow",
+			"subject": "*",
+			"resources": [
+				"dataset:_subject:*"
+			],
+			"actions": [
+				"remote:push"
+			]
+		}
+	]
+`
+	allowPushPolicy := &access.Policy{}
+	if err := json.Unmarshal([]byte(allowPushJSON), allowPushPolicy); err != nil {
+		t.Fatalf("error unmarshalling 'allowPushPolicy': %q", err)
+	}
```

Comment:
I'm a fan of defining a `mustJSON` helper function for these in tests:

```go
func mustJSON(data string, v interface) {
  if err := json.Unmarshal([]byte(data), v); err != nil {
    panic(err)
  }
}
```

```suggestion
	allowPushPolicy := &access.Policy{}
	mustJSON( `
	[
		{
			"title": "allow subject to push its own datasets",
			"effect": "allow",
			"subject": "*",
			"resources": [
				"dataset:_subject:*"
			],
			"actions": [
				"remote:push"
			]
		}
	]
`, allowPushPolicy)
```

Reply:
LOVE IT